Source/Models/SwishError.swift

```diff
@@ -0,0 +1,26 @@
+import Foundation
+import Argo
+
+public enum SwishError {
+  case SessionError(NSError)
+  case ServerError(code: Int, json: AnyObject)
+  case JSONSerializationError(NSError)
+  case ArgoError(DecodeError)
```

Comment:
That's a good start. I think it would be good to then look at each name and consider whether it's clear from just the name where the error occurred.

For example: `case Session(NSError)`. Is that wrapping an error returned from an `NSURLSession` task? It's not entirely clear.